app/models/manageiq/providers/azure/manager_mixin.rb

```diff
@@ -11,12 +11,23 @@ def connect(options = {})
 
   def verify_credentials(_auth_type = nil, options = {})
     require 'azure-armrest'
-    connect(options)
-  rescue Azure::Armrest::UnauthorizedException
-    raise MiqException::MiqHostError, _("Incorrect credentials - check your Azure Client ID and Client Key")
+    conf = connect(options)
+
+    unless subscription.blank?
+      vms = ::Azure::Armrest::VirtualMachineService.new(conf)
+      subscriptions = vms.subscriptions.map(&:subscription_id)
+      unless subscriptions.include?(subscription)
+        raise MiqException::MiqInvalidCredentialsError, _("Incorrect credentials - check your Azure Subscription ID")
+      end
+    end
+  rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
```

Comment:
Yes.

Note that what's actually happening is a fetch_subscription_id call inside the azure-armrest gem. This is not really ideal, since it relies on internal behavior (which is how we ended up here in the first place), and Bill and I have agreed that we're going to add a method to externally validate credentials in azure-armrest 0.3.x so that this is all abstracted away from us.